lib/tapioca/rbs/rewriter.rb

```diff
@@ -0,0 +1,34 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "ruby-next/language/runtime"
+
+# This module rewrites RBS comments back into Sorbet's signatures as the files are being loaded.
+# This will allow `sorbet-runtime` to wrap the methods as if they were originally written with the `sig{}` blocks.
+# This will in turn allow Tapioca to use this signatures to generate typed RBI files.
+
+# We need to include `T::Sig` very early to make sure that the `sig` method is available since gems using RBS comments
+# are unlikely to include `T::Sig` in their own classes.
+Module.include(T::Sig)
+
+module Tapioca
+  module RBS
+    # Transpiles RBS comments back into Sorbet's `sig{}` blocks
+    class Rewriter < RubyNext::Language::Rewriters::Text
+      NAME = "rbs_rewriter"
+
+      #: (String source) -> String
+      def rewrite(source)
+        # Do not try to parse files that don't have RBS comments
+        return source unless source =~ /^\s*#\s*typed: (ignore|false|true|strict|strong|__STDLIB_INTERNAL)/
+
+        context.track!(self)
+        Spoom::Sorbet::Sigs.rbs_to_rbi(source)
+      end
+    end
+  end
+end
+
+RubyNext::Language.include_patterns.clear
+RubyNext::Language.include_patterns << "**/*.rb"
```

Comment:
Why do these need to be cleared first? Does it include something other than Ruby files by default?

We could do

```suggestion
RubyNext::Language.include_patterns = ["**/*.rb"]
```

or just append if it's already empty.

Reply:
Yes, the patterns list does include other patterns by default, but yeah, agreed, we could just assign the array instead of clearing and appending.

Reply:
The writer is [not exposed](https://github.com/ruby-next/ruby-next/blob/da255b8896d3378aee844508d0e8d37c3c9980d3/lib/ruby-next/language.rb#L69).

About the clear, we _could_ keep [the default](https://github.com/ruby-next/ruby-next/blob/da255b8896d3378aee844508d0e8d37c3c9980d3/lib/uby-next.rb#L12), I'm just not taking any change in case it changes.